anoma/namada-interface
# feat: fully decode IBC assets on cosmos chains

If there are IBC assets on a Keplr account, attempt to look them up on the chain they came from and display them. For example, if there are NAM tokens on a Cosmos Hub account, display them as NAM.

Notes:
- This currently works for IBC assets from all cosmos chains and from the Namada internal devnet, but NOT any other Namada chains (e.g. housefire). Follow up issue: #1218
- I harcoded some registry entries temporarily, which should be removed once we've integrated the Namada registry. Follow up issue: #1217
- I don't know if this works for tokens that have been transferred more than once (e.g. Namada -> Osmosis -> Cosmos Hub). Follow up issue: #1219
- There is a bug where if you have the same asset from different chains (e.g. you sent NAM from both housefire and the internal devnet), only one of the balances will show. It is not so straightforward to fix, so follow up issue: #1220

### Testing

1. Configure Namadillo to use the internal devnet
2. Go to IBC Transfer -> From Namada and send some NAM to your Cosmos Hub testnet account (assuming Namadillo IBC features are enabled and Keplr is set up)
3. Go to IBC Transfer -> To Namada and select the Cosmos Hub testnet
4. In the asset list, you should see the tokens you sent listed as "Namada" and the denomination treated as NAM (e.g. if you have 1 NAM, it shows the balance as 1, not 1000000)

You can also try sending other IBC tokens (e.g. send some OSMO from Osmosis testnet to Cosmos Hub testnet) and it should also display properly.
